app/src/main/java/com/koalatea/thehollidayinn/softwareengineeringdaily/PlaybackControllerActivity.java

```diff
@@ -171,9 +171,14 @@ protected void onMediaItemSelected(MediaBrowserCompat.MediaItem item, boolean is
             MediaControllerCompat.TransportControls controls = controller.getTransportControls();
 
             if (isPlaying) {
-                controls.pause();
+                if (controller.getPlaybackState().getState() == PlaybackStateCompat.STATE_PLAYING) {
+                    controls.pause();
+                } else if (controller.getPlaybackState().getState() == PlaybackStateCompat.STATE_PAUSED) {
+                    controls.play();
+                }
             } else {
                 controls.playFromMediaId(item.getMediaId(), null);
+                mCurrentMediaId = item.getMediaId();
```

Comment:
What is the variable set for?

Reply:
It's just a variable that stores the ID of the current playing episode. We use it to compare if the selected episode is the same as the currently playing one.

Reply:
GOTcha, but where is it used in this block?

Reply:
https://github.com/SoftwareEngineeringDaily/SEDaily-Android/blob/42b22bbe72f5164fe9b4ae49b2ad5eff48ad47a6/app/src/main/java/com/koalatea/thehollidayinn/softwareengineeringdaily/podcast/PodcastDetailActivity.java#L331

Reply:
Oh wow.... I'm sorry. I didn't realize this was an assignment and not an instantiation.